src/razor/src/document/razorDocumentManager.ts

```diff
@@ -219,6 +221,26 @@ export class RazorDocumentManager implements IRazorDocumentManager {
         this.notifyDocumentChange(document, RazorDocumentChangeKind.removed);
     }
 
+    private findDocument(path: string) {
+        // This method ultimately gets called from VS Code, using file system paths, or from DevKit, which
+        // can use paths as specified in the sln file. When these don't agree, on case-insensitive operating
+        // systems, we have to be careful to match things up correctly.
+
+        if (this.platformInfo.isLinux()) {
```

Comment:
This will probably work most of the time, but its very possible to have either a case sensitive or insensitive file system on linux (and others).  I'm not sure if there is a way to generally know in node though, other than checking for the existence of the path on disk.

Reply:
Neither Roslyn or Razor seem to do anything more sophisticated than this. See https://github.com/dotnet/roslyn/blob/main/src/Compilers/Core/Portable/FileSystem/PathUtilities.cs#L675

This low down in the code, I don't think file existence is something we can rely on either - we just want to compare two document paths. VS Code will happily keep around a document, and issue LSP requests for it, if it doesn't exist on disk. Likewise Roslyn will happily answer requests about it until it gets a didClose, etc.